File: euler/002/euler_002.rb

def main()
    first = 0; second = 1; sum = 0;

    loop do
        fib = get_next_fib(first, second)
        break if fib > 4_000_000
        sum += fib.even? ? fib : 0
        first, second = second, fib
    end

    puts sum
end

def get_next_fib(first, second)
    return first + second
end

main()

File: euler/002/euler_002.js

function main(){
    first = 0, second = 1, sum = 0;

    while(true){
        fib = getNextFib(first, second);
        if(fib > 4000000){
            break;
        }

        if(fib % 2 == 0){
            sum += fib;
        }
        first = second;
        second = fib; 
    }

    console.log(sum);
}

function getNextFib(first, second){
    return first + second;
}

main();